### PR TITLE
feat: add ssl support using custom key/cert files

### PR DESCRIPTION
**Pull request by dahifi**
Someone on the discord asked for a way to run the app behind https, so I added a run config parameters to accept keyfile and cert paths, and pass them to uvicorn. 

I was unable to run commit hooks on this, nor did I test, but putting this here to see if anyone can validate or pick it up from here.

### CODE REVIEW

**Comment by puppetm4st3r** [comment]
if I add those changes to my local, I it will work? or is incomplete ? the added parameters must be provided as environment vars or running paremeters in chainlit executable ?

Best regards!

**Comment by puppetm4st3r** [comment]
in my case the reverse proxy + chainlit + serving wasm artifacts didnt liked it to iOS devices, so I have solved appliying this PR and removing the prody and works fine.... and its a clearer solution for lite infrastructure use cases.

**Comment by veeragoni** [comment]
localhost

**Comment by veeragoni** [comment]
> Thanks for your PR, there is a similar one here: #905
> 
> Do you have more background as to why folks who want to create a publicly accessible Chainlit instance without having a proxy in front of it?

localhost

File: backend/chainlit/cli/__init__.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def run_chainlit(target: str):
     host = os.environ.get("CHAINLIT_HOST", DEFAULT_HOST)
     port = int(os.environ.get("CHAINLIT_PORT", DEFAULT_PORT))
 
+    ssl_certfile = os.environ.get("SSL_CERT", None)
+    ssl_keyfile = os.environ.get("SSL_KEY", None)
```
Comment (tpatel) [inline backend/chainlit/cli/__init__.py:42]:
```suggestion
    ssl_certfile = os.environ.get("CHAINLIT_SSL_CERT", None)
    ssl_keyfile = os.environ.get("CHAINLIT_SSL_KEY", None)
```

File: backend/chainlit/cli/__init__.py [hunk 2]
```diff
@@ -126,13 +131,30 @@ async def start():
     envvar="NO_CACHE",
     help="Useful to disable third parties cache, such as langchain.",
 )
+@click.option(
+    "--ssl-cert",
+    default=None,
+    envvar="SSL_CERT",
```
Comment (tpatel) [inline backend/chainlit/cli/__init__.py:137]:
```suggestion
    envvar="CHAINLIT_SSL_CERT",
```

File: backend/chainlit/cli/__init__.py [hunk 3]
```diff
@@ -126,13 +131,30 @@ async def start():
...
+    help="Specify the file path for the SSL certificate.",
+)
+@click.option(
+    "--ssl-key",
+    default=None,
+    envvar="SSL_KEY",
```
Comment (tpatel) [inline backend/chainlit/cli/__init__.py:143]:
```suggestion
    envvar="CHAINLIT_SSL_KEY",
```

File: backend/chainlit/cli/__init__.py [hunk 4]
```diff
@@ -126,13 +131,30 @@ async def start():
...
+def chainlit_run(target, watch, headless, debug, ci, no_cache, ssl_cert, ssl_key, host, port):
     if host:
         os.environ["CHAINLIT_HOST"] = host
     if port:
         os.environ["CHAINLIT_PORT"] = port
+    if bool(ssl_cert) != bool(ssl_key):
+        raise click.UsageError("Both --ssl-cert and --ssl-key must be provided together.")
+    if ssl_cert:
+        os.environ["SSL_CERT"] = ssl_cert
+        os.environ["SSL_KEY"] = ssl_key
```
Comment (tpatel) [inline backend/chainlit/cli/__init__.py:157]:
```suggestion
        os.environ["CHAINLIT_SSL_CERT"] = ssl_cert
        os.environ["CHAINLIT_SSL_KEY"] = ssl_key
```

**Review by tpatel — CHANGES_REQUESTED** [review]
Looks good, could you add the `CHAINLIT_` prefix to env variables?
I've tried to do it myself but I don't have the rights on your fork.

**Review by tpatel — CHANGES_REQUESTED** [review]
There are two remaining suggestions that needs to be committed for me to merge this PR.

**Comment by yiouyou** [comment]
@tpatel This merge is quite useful when running the voice feature of chainlit. Could you check if it's ready to merge? If there's a simple document of --ssl-cert etc on the repo main page, that would be very useful.

Thanks!

**Comment by tpatel** [comment]
closing in favor of https://github.com/Chainlit/chainlit/pull/1047 (I've made the change there).

**Pull request closed without merging**